bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TableImpl.java

```diff
@@ -0,0 +1,127 @@
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.Optional;
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.SlingObject;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Objects.nonNull;
+
+@Model(adaptables = SlingHttpServletRequest.class,
+    adapters = {Table.class, ComponentExporter.class},
+    resourceType = TableImpl.RESOURCE_TYPE)
+
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class TableImpl extends AbstractComponentImpl implements Table {
+
+    public static final String RESOURCE_TYPE = "core/wcm/components/table/v1/table";
+    @ValueMapValue(name = "source", injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String source;
+
+    @ValueMapValue(name = "headerNames", injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String[] headerNames;
+
+    @ValueMapValue(name = "description", injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String description;
+
+    @Inject @Optional
+    private List<ResourceProcessor> resourceProcessors;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+    private List<String> formattedTableHeaderNames;
+    private List<List<String>> items;
+
+
+    @PostConstruct
+    public void initModel() {
+        formatHeaderNames();
+    }
+
+    private String getSourceResourceMimeType(Resource sourceResource) {
+        if (nonNull(sourceResource)) {
+            if (DamUtil.isAsset(sourceResource)) {
+                Asset asset = DamUtil.resolveToAsset(sourceResource);
+                return asset.getMimeType();
+            } else {
+                return StringUtils.EMPTY;
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+    /**
+     * This method formats the property name to friendly names by removing jcr: prefix. Formatted
+     * property names are used to display the headers in table.
+     */
+    private void formatHeaderNames() {
+        formattedTableHeaderNames = new ArrayList<>();
+        for (String propertyName : headerNames) {
+            if (propertyName.contains("jcr:")) {
+                formattedTableHeaderNames.add(propertyName.replace("jcr:",""));
+            } else {
+                formattedTableHeaderNames.add(propertyName);
+            }
+        }
+    }
+
+    @Override
+    public List<String> getFormattedHeaderNames() {
+        return formattedTableHeaderNames;
+    }
+
+    @Override
+    public List<List<String>> getItems() throws IOException {
+        if (nonNull(resourceProcessors)) {
```

Comment:
field `this.items` is updated on every call but not used.
Should either eliminate the object field `items` and just return directly; or, more likely, this is intended to only process the data once, in which case it should only be run if `this.items == null`

Reply:
Removed the items object. Returning empty list for negative scenarios.